R/slidingWindowFeatureCounts.R

```diff
@@ -0,0 +1,159 @@
+#'   paired-end reads. 600 by default. minFragLength and maxFragLength are only
+#'   applicable when isPairedEnd is TRUE. Note that when a fragment spans two
+#'   or more exons, the observed fragment length might be much bigger than the
+#'   nominal fragment length.
+#' @param minMQS An integer(1), giving the minimum mapping quality score a read
+#'   must satisfy in order to be counted. For paired-end reads, at least one end
+#'   should satisfy this criteria. 0 by default.
+#' @param verbose A logical(1) vector, indicating if verbose information for
+#' debugging will be generated. This may include information such as unmatched
+#' chromosomes/contigs between reads and annotation.
+#'
+#' @return A list of lists, each of the sublist contains an output of the
+#'   [Rsubread::featureCounts()] function for each type of genomic features.
+#'   For more details of each sublist, see the *value* section of the
+#'   documentation for the [Rsubread::featureCounts()] function.
+#'
+#' @importFrom Rsubread featureCounts
+#' @export
+#' @examples
+#' bamfiles <- list.files(system.file("extdata", package="NADfinder"),
+#'                        recursive=FALSE,
+#'                        pattern="*bam$",
+#'                        full=TRUE)
+#' names(bamfiles) <- c("G18", "N18")
+#' genomeSW_SAF <-
+#'     generateGenomeSAF(bamfile = bamfiles[1],
+#'                       windowSize = 50000L,
+#'                       step = 5000L,
+#'                       chrExcluded = c("chrM", "MT"))
+#'
+#' counts <- swFeatureCounts(bamfiles = bamfiles,
+#'                           saf = genomeSW_SAF,
+#'                           isPaired = TRUE)
+#'
+
+swFeatureCounts <- function(bamfiles = NULL,
+                            saf = NULL,
+                            isPairedEnd = TRUE,
+                            threads = 1L,
+                            checkFragLength = FALSE,
+                            requireBothEndsMapped = FALSE,
+                            minFragLength = 50L,
+                            maxFragLength = 1000L,
+                            minMQS = 0,
+                            verbose = FALSE)
+    {
+       if (any(!file.exists(bamfiles)))
+       {
+           stop("Not all BAM files exist!")
+       } else if (is.null(names(bamfiles))) {
+           stop("BAM files must be a named vector of character!")
+       }
+       saf_colnames <- c("GeneID", "Chr", "Start", "End", "Strand")
+       if (is.null(saf))
+       {
+           stop("saf is required!")
+       }  else if (!is.data.frame(saf) ||
+                   !all(saf_colnames == colnames(saf[[1]]))) {
+           stop("saf must be a data frame!")
+       }
+       res <- featureCounts(
+           files = bamfiles,
+           # annotation
+           annot.ext = saf,
+           isGTFAnnotationFile = FALSE,
+           # overlap between reads and features
+           allowMultiOverlap = TRUE,
```

Comment:
I did not know they add this prarameter now. Did you test this? Does it really work as they addressed for both bin?

Reply:
Yes. I used your subsampled bam files to compare the tileCount result with the featureCounts result. They are very similar. I believe the minor difference is due to the different way to handle multi-mapping reads. I chose not to count multi-mapping reads at all. I am not sure how summarizeOverlaps handle multi-mapping reads.